compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2505,3 +2505,10 @@ import transform.SymUtils._
           |Inlining such definition would multiply this footprint for each call site.
           |""".stripMargin
   }
+
+  class OpenModifierRedundantForAbstractClassOrTrait(cls: untpd.TypeDef, isTrait: Boolean)(using Context)
```

Comment:
Instead of defining a new message class, it might be possible to generalize the existing ModifierRedundantForObjects class to a ModifierRedundant that can be used for non-classes too.

Reply:
I would need to disambiguate between abstract/inner classes, traits and objects. Should this be done on an `untpd.Tree` or is there a more suitable abstraction?

Reply:
Looks like the cls argument isn't used here and could just be removed

Reply:
It would be nice to keep the explain-method from `ModifierRedundantForObjects` around and generalise it. I could pass the information from the report site as a String but I think it'd richer information would be preferable, right?

Reply:
I patched the functionality a bit. It would be nice if the recommended solution in `explain` kept the template (or lack thereof) around, but for large defs that would be spammy unless truncated.